pandas/tests/groupby/test_rank.py

```diff
@@ -290,3 +290,17 @@ def test_rank_empty_group():
     result = df.groupby(column).rank(pct=True)
     expected = DataFrame({"B": [0.5, np.nan, 1.0]})
     tm.assert_frame_equal(result, expected)
+
+
+def test_rank_zero_div():
+    # GH 23666
+    df = pd.DataFrame({
+        "A": [1, 1, 1, 2, 2, 2],
+        "B": [1, 1, 1, 1, 2, 2],
+        "C": [1, 2, 1, 1, 1, 2]
+    })
+
+    try:
```

Comment:
You don't need a try except here - exceptions will be propagated naturally and cause a test failure

Reply:
Would rather just be explicit about the expected result and test for that